### PR TITLE
Removes MetaStation from Map Vote

### PR DESCRIPTION
**Pull request by matttheficus**
## What Does This PR Do
Removes MetaStation (NSS Cerebron) from the Map Vote options. Leaves the Map available for admins to force if need-be.

## Why It's Good For The Game
MetaStation is a rather unpopular and unfinished map in comparison to our other two mainstays. Until its either improved or a proper replacement is found, this PR will remove it from being picked.

Some data for this decision:
Of the 305 responses to the Community Survey on their least favorite map was either listed or included "metastation" 139 times.

 **(139/305 - ~45%)**

In comparison, for favorite station, its at 35/305.

**(35/305 - ~11%)**

(mind you a large quantity of players have never even seen Metastation considering its very rarely voted as well as a section of the responses being n/a or uncaring).

In regards to my personal anecdotes regarding MetaStation:
1. Survey seems to show a large quantity of players dislike the station.
2. In-game it is meme'd to hell and back about being an unpopular station that causes players to actively log off/skip the round.
3. There's a large quantity of flaws in its design from a gameplay perspective (1x wide maints, missing QoL bits other stations have, etc.) that other stations have either had fixed or lack.
4. its literally Box, turned, and has no "gimmick" that isn't done better by the two mainstays (Box and Delta).
5. Mappers don't seem to want to touch it with a 10 foot pole for either above reasons or personal dislike for it (note this is only from comments I've seen in the past).

Unsure if this is GBP-affecting due to it being a somewhat configuration change.

## Changelog
:cl:
del: Removes MetaStation from the Map Vote pool.
/:cl:

### CODE REVIEW

**Comment by matttheficus** [comment]
If this is too snowflakey, I can look into changing this to a config-based thing though I would need AA's permission for that I suppose.

File: code/controllers/subsystem/vote.dm [hunk 1]
```diff
@@ -180,7 +180,7 @@ SUBSYSTEM_DEF(vote)
 			if("map")
 				// Find target map.
 				var/datum/map/top_voted_map
-				for(var/x in subtypesof(/datum/map))
+				for(var/x in subtypesof(/datum/map) - /datum/map/metastation) //remove this Meta removal once we find a fix to Meta/another Map. - MattTheFicus
```
Comment (AffectedArc07) [inline code/controllers/subsystem/vote.dm:183]: This is incredibly snowflake and should be done on the map datum itself with a voteable var.

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

**Comment by AffectedArc07** [comment]
Really not in favour of this. It seems like an underhanded way to remove something without needing to muster up the support for a full removal PR. 

Take gateway missions for example, half of those are disabled instead of being fully removed because that would require more support behind a PR than something like this.


On top of that, if we take meta out the pool, that's just going to incentivise people to not care about it like what happened for the 5 years it's already been in the codebase before I added map rotation.

TLDR: 👎

**Comment by BakKeener** [comment]
The map still needs major updating, I looked at the SM and there is vents under the SM pipes by default which could cause atmos issues by bluespacing in O2. I think perma still doesn't have a cryo sleeper, I actually like dorms but Cryogenics is not obvious enough. There is also a bunch of medbay of medbay vents/scrubbers that aren't connected to any pipenets at all according to Marm.

**Comment by matttheficus** [comment]
> Really not in favour of this. It seems like an underhanded way to remove something without needing to muster up the support for a full removal PR.

It's been brought up before to Staff and there was a general feel of "yes Meta is bad". I can attempt to go to more of Leadership in regards to a proper removal (or implementation of a voteable var).

I know that Mochi has stated not minding seeing Meta go and Denth has stated that they dislike its awful design. General Staff consensus seems to be similar (or at least those who involved themselves in the discussion). Playerbase input has been taken from the Survey and from general conversations I've seen on Discord.

> On top of that, if we take meta out the pool, that's just going to incentivize people to not care about it like what happened for the 5 years it's already been in the codebase before I added map rotation.

If this was the case, people would have fixed it already. Mappers don't seem to want to give Meta the time of day due to its flaws requiring a complete remap of the whole Station. As of now there is a noticeable drop off of players any time its voted in. With reports that people are also just voting Meta and logging off afterwards, it's even more bothersome.

File: code/controllers/subsystem/vote.dm (hunk 1 above)
Comment (matttheficus) [reply code/controllers/subsystem/vote.dm]: Desnowflake'd via datum variable.

**Comment by matttheficus** [comment]
De-snowflaked and moved to the map/datum/ as a variable `voteable`.

File: code/controllers/subsystem/vote.dm [hunk 2]
```diff
@@ -239,6 +239,10 @@ SUBSYSTEM_DEF(vote)
 				question = "Map for next round"
 				for(var/x in subtypesof(/datum/map))
```
Comment (Sirryan2002) [inline code/controllers/subsystem/vote.dm:240]: couldn't this just be 
```dm
var/datum/map/M in subtypesof(/datum/map)

Comment (AffectedArc07) [reply code/controllers/subsystem/vote.dm:240]: I think so? I did it like this before I was using initial() for stuff so it might work.

**Comment by GDNgit** [comment]
Putting in my opinion on this topic
Strongly against this idea, simply disabling the map voting feature will make this an extremely temporary change. I cannot see a future where meta would not be readded on the basis that it's simply in the codebase, and due to it not being in rotation, improved into a form that is suitable for this server. 

Meta itself is in such an unsuitable state for play that a proper overhaul would leave it in a state almost totally different than the original map itself, therefore in the eyes of many making a redesign almost entirely pointless. Removal is justified on merit of design IMHO.

This is a bit of an off-topic to the actual PR, but this is a precedent that I feel is extraordinarily dangerous to set for the community at large, the idea that you can "simply dummy out something and then fix it later" almost always ends in disaster, and functions as a removal simply done in an underhanded manner. It highly encourages stagnation and procrastination as nobody has /any/ incentive to fix the problem at hand, and the basic logic behind the idea can be applied to anything slightly unbalanced.

File: code/controllers/subsystem/vote.dm (hunk 2 above)
Comment (SteelSlayer) [reply code/controllers/subsystem/vote.dm:240]: Shouldn't because `subtypesof`/`typesof` returns a list of types, while `var/datum/map/M` is searching for an instance.

**Comment by Fox-McCloud** [comment]
Not in favor of this either.

👎

**Comment by Denthamos** [comment]
Leaving the map in and turning off the vote for it really makes no sense as then people making changes/updates that effect delta/cyb also have to update meta still.  If such a thing is going to be done it should just be the full removal of meta.

**Comment by matttheficus** [comment]
Overall seems Leadership's against it, closing so that other map things can be unhindered.
